ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/launcher/TracingBlock.java

```diff
@@ -58,6 +63,55 @@
 
 public class TracingBlock {
 
+	/**
+	 * This class encapsulates all calls to the tracing options manager and runs them using a
+	 */
+	private static record TracingOptionsManagerDelegate(IRunnableContext fRunnableContext) {
+
+		Map<String, String> getTracingTemplateCopy() {
+			return runShowingProgress(
+					(tracingOptionsManager, monitor) -> tracingOptionsManager.getTracingTemplateCopy(monitor));
+		}
+
+		public Map<String, String> getTracingOptions(Map<String, String> options) {
+			return runShowingProgress(
+					(tracingOptionsManager, monitor) -> tracingOptionsManager.getTracingOptions(options, monitor));
+		}
+
+		public Map<String, String> getTemplateTable(String pluginId) {
+			return runShowingProgress(
+					(tracingOptionsManager, monitor) -> tracingOptionsManager.getTemplateTable(pluginId, monitor));
+		}
+
+		private Map<String, String> runShowingProgress(
+				BiFunction<TracingOptionsManager, IProgressMonitor, Map<String, String>> task) {
+			try {
+				String taskName = PDEUIMessages.TracingBlock_initializing_tracing_options;
+				final Map<String, String> result = new HashMap<>();
+
+				// due to a bug
+				// (https://github.com/eclipse-platform/eclipse.platform/issues/769),
+				// the task needs to be forked otherwise the UI
+				// does not show the progress indicator
+				fRunnableContext.run(true, false, monitor -> {
+					SubMonitor subMonitor = SubMonitor.convert(monitor, taskName, 1);
+					result.putAll(task.apply(PDECore.getDefault().getTracingOptionsManager(), subMonitor.split(1)));
+				});
+
+				return result;
+			} catch (InvocationTargetException | InterruptedException e) {
+				PDEPlugin.logException(e);
+			}
+
+			// This should never happen since the task can not be canceled
+			// (yet). If it does happen then some exception occurred while
+			// calling the TracingOptionsManager and all bets are off (i.e. NPEs
+			// will follow)
+			return null;
```

Comment:
Good point, I changed it.

Better to have a descriptive exception than to let it throw a follow-up NPE. 

For the record: I will tackle the issue of showing the exception to the user in the UI when I make the task cancelable (in another PR) since then it will be much more likely that an exception happens (specifically: [an `OperationCanceledException` wrapped inside a `InterruptedException`](https://github.com/eclipse-platform/eclipse.platform.ui/pull/1259))